bool save_hole
---
bool success
---